internal/provider/resource_apply_module.go

```diff
@@ -387,43 +447,62 @@ func (t *applyModuleResource) doRun(ctx context.Context,
 	})
 	if err != nil {
 		diags.AddError("Failed to create run", err.Error())
-		return diags
+		return nil, diags
 	}
 
 	if err = t.waitForJobCompletion(ctx, createdRun.Plan.CurrentJobID); err != nil {
 		diags.AddError("Failed to wait for plan job completion", err.Error())
-		return diags
+		return nil, diags
 	}
 
 	plannedRun, err := t.client.Run.GetRun(ctx, &sdktypes.GetRunInput{ID: createdRun.Metadata.ID})
 	if err != nil {
 		diags.AddError("Failed to get planned run", err.Error())
-		return diags
+		return nil, diags
 	}
 
 	// If the plan fails, both plannedRun.Status and plannedRun.Plan.Status are "errored".
 	// If the plan succeeds, plannedRun.Status is "planned",
 	// while plannedRun.Plan.Status is "finished".
 	//
-	if !strings.HasPrefix(string(plannedRun.Status), "planned") {
-		diags.AddError("Plan failed", string(plannedRun.Status))
-		return diags
-	}
-	if plannedRun.Plan.Status != "finished" {
-		diags.AddError("Plan failed", string(plannedRun.Plan.Status))
-		return diags
+	var planDetail string
+	switch {
+	case (plannedRun.Status != sdktypes.RunPlanned) && (plannedRun.Status != sdktypes.RunPlannedAndFinished):
```

Comment:
This is what the code looks like in main from before this story was started:
![image](https://github.com/martian-cloud/terraform-provider-tharsis/assets/86491630/5424a041-04b5-467d-bf87-b74cc1402360)
Are you saying it was incorrect or at least unnecessary for both of those checks to have been there?

Being as this is a bug fix story, I'm trying to retain as much of the original intent as possible, because there are likely subtle issues I don't understand well enough to rewrite the code.  Which of those checks would you prefer to keep?

Reply:
Only the check for the plan status should be needed since the plan will be set to `errored` if it fails.

Reply:
Done.